juriscraper/opinions/united_states/state/ny.py

```diff
@@ -6,132 +6,117 @@
-        self.url = "http://www.nycourts.gov/ctapps/Decisions/{year}/{mon}{yr}/{month}{yr}.html".format(
-            year=today.year,
-            yr=today.strftime("%y"),
-            mon=today.strftime("%b"),
-            month=today.strftime("%B"),
-        )
+        self.court = "Court of Appeals"
         self.court_id = self.__module__
+        self.url = "https://iapps.courts.state.ny.us/lawReporting/Search?searchType=opinion"
+        self._set_parameters()
+        self.expected_content_types = ["application/pdf", "text/html"]
+        self.make_backscrape_iterable(kwargs)
         set_api_token_header(self)
 
-    def _make_html_tree(self, text):
-        return get_html5_parsed_text(text)
-
-    def _get_case_names(self):
-        path = f"{self.FOUR_CELLS_SUB_PATH} and {self.DOWNLOAD_URL_SUB_PATH}]"
-        case_names = []
-        for element in self.html.xpath(path):
-            case_name_parts = []
-            for t in element.xpath("./td[3]/p/font/text()"):
-                if t.strip():
-                    case_name_parts.append(t)
-            if not case_name_parts:
-                # No hits for first XPath, try another that sometimes works.
-                for t in element.xpath("./td[3]//text()"):
-                    if t.strip():
-                        case_name_parts.append(t)
-            if case_name_parts:
-                case_names.append(", ".join(case_name_parts))
-        return case_names
-
-    def _get_download_urls(self):
-        return self.html.xpath(
-            f"{self.FOUR_CELLS_SUB_PATH}]/{self.DOWNLOAD_URL_SUB_PATH}"
+    def _set_parameters(
+        self,
+        start_date: Optional[date] = None,
+        end_date: Optional[date] = None,
+    ) -> None:
+        """Set the parameters for the POST request.
+
+        If no start or end dates are given, scrape last month.
+        This is the default behaviour for the present time scraper
+
+        :param start_date:
+        :param end_date:
+
+        :return: None
+        """
+        self.method = "POST"
+
+        if not end_date:
+            end_date = date.today()
+            start_date = end_date - timedelta(days=30)
+
+        self.parameters = {
+            "rbOpinionMotion": "opinion",
+            "Pty": "",
+            "and_or": "and",
+            "dtStartDate": start_date.strftime("%m/%d/%Y"),
+            "dtEndDate": end_date.strftime("%m/%d/%Y"),
+            "court": self.court,
+            "docket": "",
+            "judge": "",
+            "slipYear": "",
+            "slipNo": "",
+            "OffVol": "",
+            "Rptr": "",
+            "OffPage": "",
+            "fullText": "",
+            "and_or2": "and",
+            "Order_By": "Party Name",
+            "Submit": "Find",
+            "hidden1": "",
+            "hidden2": "",
+        }
+
+    def _process_html(self):
+        for row in self.html.xpath(".//table")[-1].xpath(".//tr")[1:]:
```

Comment:
Should we test for the more than 500 results error page? Test for that xpath and send a `logger.error` call to Sentry so we can now of missing data and schedule a backscrape with variable days interval. I remember this happening at least once, I left a comment in one of the NY issues